src/view.c

```diff
@@ -611,6 +612,41 @@ view_get_edge_snap_box(struct view *view, struct output *output,
 	return dst;
 }
 
+/* Move and resize view based on percent output local values in rel_box */
+void
+view_rel_move_resize_to(struct view *view, struct wlr_box *rel_box)
```

Comment:
How about re-writing this function to be `absolute`, but prior to calling it do a `resolve_size()` to translate width/height expressed as percentages or fractions into actual pixels.

A `resolve_size()` could honour `rc.gap`

```
width_ratio = 0.5;  /* calculate by resolve_size() */
nr_gaps = (1/width_ratio + 1)
width = (output->width - nr_gaps) * width_ratio;
```
(Might have got that slightly wrong... but hopefully you know where I'm going)

Reply:
hm.. basically this is already something like that. At the end we call the absolute `view_move_resize()`.
About the gaps I am not sure they would fit here at all.. If a users wants gaps between the tiled windows they could offset the `x` and `y` variants by a percent.

Currently this prototype only supports integers for the percentages (as we internally use the wlr_box for storing) but I guess it could be replaced by `struct wlr_fbox` instead.

Reply:
I wanna gaps! Actually, it would be nice to (optionally) respect gaps in maximized mode too (however, with MoveResizeTo I can do it myself of course).